tests/test_storage.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+"""
+This is a unit test for representing storage in the MESSAGEix model, and
+testing the functionality of storage equations. The workflow is as follows:
+    - building a stylized MESSAGEix model
+    - adding seasonality and modifying parameters accordingly
+    - adding storage implementation
+    - testing storage functionality and equations
+
+"""
+
+from message_ix import Scenario
+
+# A flag to show if storage variables/parameters are added to ixmp java backend
+storage_in_java = False
+
+
+# A function for generating a simple MESSAGEix model
+def model_setup(scen, years):
+    scen.add_set('node', 'node')
+    scen.add_set('commodity', 'electr')
+    scen.add_set('level', 'level')
+    scen.add_set('year', years)
+    scen.add_set('type_year', years)
+    scen.add_set('technology', ['tec1', 'tec2'])
+    scen.add_set('mode', 'mode')
+    output_specs = ['node', 'electr', 'level', 'year', 'year']
+    dict_var_cost = {'tec1': 1, 'tec2': 2}
+
+    for yr in years:
+        scen.add_par('demand', ['node', 'electr', 'level', yr, 'year'],
+                     1, 'GWa')
+        for t in dict_var_cost.keys():
+            tec_specs = ['node', t, yr, yr, 'mode']
+            scen.add_par('output', tec_specs + output_specs, 1, 'GWa')
+            scen.add_par('var_cost',
+                         tec_specs + ['year'], dict_var_cost[t], 'USD/GWa')
+
+
+# A function for adding sub-annual time steps to a MESSAGEix model
+def add_seasonality(scen, seasons_dict):
```

Comment:
@behnam2015 do you think it would be worth it to make this a top-level `message_ix.Scenario` function? Right now we have an [`add_horizon()`](https://github.com/iiasa/message_ix/blob/master/message_ix/core.py#L209) function which helps you set up a model. I think it would be great to have this somewhere callable by others as well! It would need to do all the required work though, so something like the `year_to_time` below as well.

Reply:
@ Matt, thanks for the suggestion! I agree it will be a good addition. @adrivinca also suggested that the order of each time step can be specified through a set. Now, we specifiy the time order through a parameter: `time_seq`. So, `add_season` or something similar can be optionally used to wrap the information related to 
1. a subannual time step (set `time`)
2. hirerachy of that time step (set `lvl_temporal` and `map_temporal_hierarchy`)
3. Duration of that time step (parameter `duration_time`)
4. The order of that time step in a year (in this PR, parameter `time_seq`)